public.yaml

```diff
@@ -61,6 +61,35 @@ metadata:
         ]
       },
       {
+        "name": "salt-minion-key-generation",
+        "image": "sles12/salt-master:2016.11.4",
+        "command": ["sh", "-c", "umask 377;
+                                 salt-key --gen-keys=ca --gen-keys-dir /tmp;
+                                 salt-key --gen-keys=admin --gen-keys-dir /tmp;
```

Comment:
What happens after a restart? Should we only separate those in two blocks depending on each one for more safety?

```
mkdir /salt-master-pki/minions/;
if [ ! -f /salt-admin-minion-pki/minion.pub ] || [ ! -f /salt-admin-minion-pki/minion.pem ]; then
  salt-key --gen-keys=admin --gen-keys-dir /tmp;
  cp /tmp/admin.pub /salt-master-pki/minions/admin;
  mv /tmp/admin.pub /salt-admin-minion-pki/minion.pub;
  mv /tmp/admin.pem /salt-admin-minion-pki/minion.pem;
fi
if [ ! -f /salt-ca-minion-pki/minion.pub ] || [ ! -f /tmp/ca.pem /salt-ca-minion-pki/minion.pem ]; then
  cp /tmp/ca.pub /salt-master-pki/minions/ca;
  salt-key --gen-keys=ca --gen-keys-dir /tmp;
  mv /tmp/ca.pub /salt-ca-minion-pki/minion.pub;
  mv /tmp/ca.pem /salt-ca-minion-pki/minion.pem;
fi
```